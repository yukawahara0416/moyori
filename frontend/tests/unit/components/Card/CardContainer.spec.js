import { shallowMount } from '@vue/test-utils'
import Component from '@/components/Card/CardContainer.vue'

let wrapper
let propsData

beforeEach(() => {
  propsData = {
    spots: [{ data: { id: 1 } }, { data: { id: 2 } }]
    // type: 'map'
  }

  wrapper = shallowMount(Component, {
    propsData
  })
})

// afterEach(() => {
//   wrapper.destroy()
// })

describe('props', () => {
  it('spots', () => {
    expect(wrapper.props().spots).toStrictEqual(propsData.spots)
    expect(wrapper.props().spots instanceof Array).toBe(true)
  })
  // it('type', () => {
  //   expect(wrapper.props().type).toStrictEqual(propsData.type)
  //   expect(typeof wrapper.vm.$props.type).toBe('string')
  // })
})

// describe('computed', () => {
//   it('empties', () => {})
// })

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
