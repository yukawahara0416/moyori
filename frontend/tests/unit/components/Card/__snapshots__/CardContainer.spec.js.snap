// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-row-stub
  align="stretch"
  class="ma-0"
  justify="space-around"
  tag="div"
>
  <v-col-stub
    class="mt-2"
    cols="12"
    lg="4"
    sm="6"
    tag="div"
  >
    <card-container-frame-stub
      id="0"
      spot="[object Object]"
      type="map"
    />
  </v-col-stub>
  <v-col-stub
    class="mt-2"
    cols="12"
    lg="4"
    sm="6"
    tag="div"
  >
    <card-container-frame-stub
      id="1"
      spot="[object Object]"
      type="map"
    />
  </v-col-stub>
   
  <v-col-stub
    class="ma-0 pa-0"
    cols="12"
    lg="4"
    sm="6"
    tag="div"
  >
    <card-container-fill-stub />
  </v-col-stub>
  <v-col-stub
    class="ma-0 pa-0"
    cols="12"
    lg="4"
    sm="6"
    tag="div"
  >
    <card-container-fill-stub />
  </v-col-stub>
</v-row-stub>
`;
